src/endian.h

```diff
@@ -2,6 +2,7 @@
 #define H_CARANO_APP_ENDIAN
 
 #include <stdint.h>
+#include "assert.h"
```

Comment:
Hmm, why is this assert needed?

Reply:
I added a STATIC assert below in the previous PR about casting. What is weird is, that master branch compiles without problems. I would like to look into that